scripts/random-responses.js

```diff
@@ -0,0 +1,95 @@
+const fs = require('fs');
+const configs = JSON.parse(
+  fs.readFileSync('config/slack-random-response.json')
+);
+
+const cachedRequests = {};
+
+const getResponses = async (robot, config) => {
+  if (config.responseList) {
+    return config.responseList;
+  }
+
+  if (config.responseUrl) {
+    if (cachedRequests[config.responseUrl]) {
+      const cached = cachedRequests[config.responseUrl];
+      if (Date.now() < cached.expiry) {
+        return cached.value;
+      }
+    }
+
+    return new Promise(resolve => {
+      robot
+        .http(config.responseUrl)
+        .header('User-Agent', '18F-bot')
+        .get()((err, res, body) => {
+        cachedRequests[config.responseUrl] = {
+          expiry: Date.now() + 60000, // five minutes
+          value: JSON.parse(body)
+        };
```

Comment:
I don't think I'm clear on the 5 minutes caching here. What are we doing?

Reply:
Comments incoming, but the idea is not to make the network hit on every single response.  Instead, it caches off the network result for five minutes before re-refetching it.

Reply:
So does that mean if call dag fact twice in a five minute window it'll give the same response, or that it only fetches the json every 5 max?

Reply:
The latter.  You'll still get a random response, but it'll only fetch the JSON every five minutes.